File: BackEnd_PersonalBlogWeb/Models/Post.cs
using System;
using System.Collections.Generic;

namespace Project_PRN232_PersonalBlogWeb.Models;

public partial class Post
{
    public int Id { get; set; }

    public string Title { get; set; } = null!;

    public string Content { get; set; } = null!;

    public int? AuthorId { get; set; }

    public int? CategoryId { get; set; }

    public DateTime CreatedDate { get; set; }

    public string? CoverImage { get; set; }

    public byte Status { get; set; }

    public virtual User? Author { get; set; }

    public virtual Category? Category { get; set; }

    public virtual ICollection<Comment> Comments { get; set; } = new List<Comment>();

    public virtual ICollection<Like> Likes { get; set; } = new List<Like>();
}

File: PersonalBlogWeb_FE/src/components/types/post.ts
// types/post.ts
// export type Post = {
//   id: number;
//   title: string;
//   content: string;
//   coverImage?: string;
//   createdDate: string;
//   categoryName: string;
//   likeCount: number;
//   commentCount: number;
//   author: {
//     authorID: number;
//     authorName: string;
//     authorAvatar: string;
//   };
// };

export type Post = {
  id: number;
  title: string;
  content: string;
  coverImage?: string;
  createdDate: string;
  categoryName: string;
  authorName: string;
  authorAvatar: string;
  likeCount: number;
  commentCount: number;
};

export type BlogPostCardProps = {
  id: number;
  coverImage: string;
  categoryName: string;
  title: string;
  content: string;
  authorName: string;
  authorAvatar: string;
  createdDate: string;
  likeCount?: number;
  commentCount?: number;
};

export type PostDetailProps = {
  id: number;
  title: string;
  content: string;
  coverImage?: string;
  createdDate: string;
  categoryName: string;
  likeCount: number;
  commentCount: number;
  author: {
    authorID: number;
    authorName: string;
    authorAvatar: string;
  };
  comments: {
    id: number;
    content: string;
    author: {
      authorID: number;
      authorName: string;
      authorAvatar: string;
    };
  }[];
};
